x-controller.js

```diff
@@ -1,3 +1,5 @@
+/** @import { XModel } from './x-model.js' */
```

Comment:
If a file called `./x-model.d.ts` exists here — TypeScript bails out. I sorta still feel like this is a “bug”, but it also seems like the maintainers suggest that this is just the way it is…